code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm

```diff
@@ -172,7 +166,20 @@
 		C.name = "[name]"
 		C.pixel_x = rand(-8, 8)
 		C.pixel_y = -16
-		reagents.trans_to(C,15)
+		reagents.trans_to(C, 30)
+		if(reagents)
+			reagents.clear_reagents()
+		C.update_icon()
+
+	else if(href_list["createwaffle"])
+		var/name = generate_name(reagents.get_master_reagent_name())
+		name += " Waffle Cone"
+		var/obj/item/reagent_containers/food/snacks/icecream/icecreamwafflecone/C
+		C = new/obj/item/reagent_containers/food/snacks/icecream/icecreamwafflecone(loc)
```

Comment:
```suggestion
		C = new /obj/item/reagent_containers/food/snacks/icecream/icecreamwafflecone(loc)
```

If you want, I think you can also just do `new(loc)` instead

Reply:
that's...new to me. Like, just straight up`C = new(loc)`?

Reply:
Yeah, I think calling an empty new() on the RHS automatically uses the type of the LHS

Reply:
Yes, lewcc is right. No need for duplicate definitions